Sources/MapboxCoreNavigation/CoreNavigationNavigator.swift

```diff
@@ -181,6 +196,15 @@ class Navigator {
         let versionsObserver = NavigatorFallbackVersionsObserver()
         navigatorFallbackVersionsObserver = versionsObserver
         navigator.setFallbackVersionsObserverFor(versionsObserver)
+        
+        if NavigationSettings.shared.alternativeRoutesOptions.enabled {
+            let alternativesObserver = NavigatorRouteAlternativesObserver()
+            navigatorAlternativesObserver = alternativesObserver
+            navigator.getRouteAlternativesController().addObserver(for: alternativesObserver)
```

Comment:
I found that [tests are failing](https://app.circleci.com/pipelines/github/mapbox/mapbox-navigation-ios/7930/workflows/16c8ec00-68d4-4e6b-9fa6-5eb6eedf2b26/jobs/68040) due to this observer setup. Even putting an "empty" observer which does nothing results in tests fail. It seems that doing so leads to severe performance degradation because most of the tests are failing with timeouts. Even straightforward ones like start/stop history recording exceed timeout of 2 seconds. Same code run on NN v94.0 worked fine, so I assume something was changed since then. @mapbox/navnative  could you please check?

Reply:
Does this exact line takes over 2 seconds? Or is it an observation about the whole test?

Reply:
At least locally on my machine, this particular line is not halting, but tests are still too long. It seems that is also matters that during our test run, we init/deallocate Navigator many times and thus set/reset observers too. For example if I run each of the failing tests individually - it passes, but running entire test suit - not. Usually this indicates that tests are not well isolated, but same tests using NN v94 worked fine.

Reply:
@DenisPryt could you work with @Udumft together to unblock it?